### PR TITLE
[ADD] Bookable validation

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -175,6 +176,28 @@ def _onchange_resource_type(self):
         self.room_id = None
         self.equipment_id = None
 
+    @api.onchange('room_id')
+    def verify_room_bookable(self):
+        if self.room_id:
```
Comment (jananjoy) [inline project_event/models/project_task.py:192]: @eilstWhat is the validation message "if self.room_id; " condition fails Or if this condition is going to true always what is the use of checking this condition.

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -175,6 +176,28 @@ def _onchange_resource_type(self):
...
+                raise ValidationError(
+                        _(
+                            'This room is not bookable'
+                        )
+                    )
+                self.room_id = None
+
+    @api.onchange('equipment_id')
+    def verify_equipment_bookable(self):
+        if self.equipment_id:
```
Comment (jananjoy) [inline project_event/models/project_task.py:201]: @eilstWhat is the validation message "if self.equipment_id: " condition fails Or if this condition is going to true always what is the use of checking this condition.

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -175,6 +176,28 @@ def _onchange_resource_type(self):
...
+            if not self.equipment_id.is_bookable:
+                raise ValidationError(
+                        _(
+                            'This resource is not bookable'
+                        )
+                    )
+                self.equipment_id = None
```
Comment (jananjoy) [inline project_event/models/project_task.py]: @eilst: Better to move this statement   self.equipment_id = None before raising the validation error message

**Review by jananjoy — CHANGES_REQUESTED** [review]
You have a few  change requests

File: project_event/models/project_task.py (hunk 1 above)
Comment (eilst) [reply project_event/models/project_task.py:192]: this is a condition to catch the error, so the else condition is implicit, to pass

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -175,6 +176,26 @@ def _onchange_resource_type(self):
         self.room_id = None
         self.equipment_id = None
 
+    @api.onchange('room_id')
+    def verify_room_bookable(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:191]: @eilst def onchange_room_id

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -175,6 +176,26 @@ def _onchange_resource_type(self):
...
+        if self.room_id:
+            if not self.room_id.is_bookable:
+                raise ValidationError(
+                        _(
+                            'This room is not bookable'
+                        )
+                    )
+                
+    @api.onchange('equipment_id')
+    def verify_equipment_bookable(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:200]: @eilst def onchange_equipment_id

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

**Review by rimbendhaou — APPROVED** [review]

**Review by jananjoy — APPROVED** [review]

**Pull request merged**